crates/hir-ty/src/next_solver/ty.rs

```diff
@@ -39,6 +39,12 @@ pub enum BoundTyKind {
 #[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Debug)]
 pub struct ErrorGuaranteed;
 
+impl Ty {
+    pub fn error() -> Self {
+        with_db_out_of_thin_air(|db| db.intern_rustc_ty(InternedTy(TyKind::Error(ErrorGuaranteed))))
```

Comment:
this should probably take an `interner` and use `interner.mk_ty` so we don't need to change as many places when we switch to new salsa interning

Reply:
I meant to remove that and use the `Ty` trait method, `new_error`. Same for the `Const`.

But I missed to do that, my bad.

Reply:
Nvm, I just remembered, the function where I was using the error doesn't have access to interner. Is there a way to access it like the DB?

Reply:
I am hoping we won't need `with_db_out_of_thin_air` with the new Salsa interning at all. Let's maybe keep this as a todo for later.

Reply:
Got it, by new interning you mean the Salsa 3.0 work?

Reply:
Yeah